projects/kit/components/input-month-range/input-month-range.directive.ts

```diff
@@ -10,16 +20,53 @@ import {TuiInputMonthRangeComponent} from './input-month-range.component';
             provide: TUI_TEXTFIELD_HOST,
             useExisting: forwardRef(() => TuiInputMonthRangeDirective),
         },
+        TuiDestroyService,
     ],
 })
-export class TuiInputMonthRangeDirective extends TuiAbstractTextfieldHost<TuiInputMonthRangeComponent> {
+export class TuiInputMonthRangeDirective
+    extends TuiAbstractTextfieldHost<TuiInputMonthRangeComponent>
+    implements DoCheck
+{
+    private readonly value$ = new Subject<TuiMonthRange | null>();
+
+    private localizedValue: [string, string] = ['', ''];
+
+    constructor(
+        @Inject(AbstractTuiControl) host: TuiInputMonthRangeComponent,
+        @Inject(TUI_MONTH_FORMATTER)
+        formatter: TuiHandler<TuiMonth | null, Observable<string>>,
+        @Self() @Inject(TuiDestroyService) destroy$: Observable<unknown>,
+    ) {
+        super(host);
+
+        this.value$
+            .pipe(
+                distinctUntilChanged(),
+                switchMap((value: TuiMonthRange | null) =>
+                    combineLatest([
+                        formatter(value && value.from),
+                        formatter(value && value.to),
+                    ]),
+                ),
+                takeUntil(destroy$),
+            )
+            .subscribe(localizedValue => {
+                this.localizedValue = localizedValue;
```

Comment:
**nit**

Can `@HostListener('change'...)` do the same thing (without subscription) ?

Reply:
We cannot because formatting relies on i18n which is `Observable`.